contracts/crowdfund/Crowdfund.sol

```diff
@@ -479,24 +494,35 @@ abstract contract Crowdfund is Implementation, ERC721Receiver, CrowdfundNFT {
         }
     }
 
+    function _getCurrentContribution(address contributor) internal view returns (uint256 totalEthContributed) {
+    Contribution[] memory contributions = _contributionsByContributor[contributor];
+    uint256 numContributions = contributions.length;
+    for (uint256 i; i < numContributions; ++i) {
+        Contribution memory c = contributions[i];
+        totalEthContributed += c.amount;
+        }
+    }
+
     function _getFinalContribution(
         address contributor
     ) internal view returns (uint256 ethUsed, uint256 ethOwed, uint256 votingPower) {
         uint256 totalEthUsed = _getFinalPrice();
         {
             Contribution[] memory contributions = _contributionsByContributor[contributor];
+            //variable for previousTotalAmount - totalContributionsWithdrawn
+            //previousTotalWithdrawn - totalWithdrawn(current) = Amount that needs to be subtracted
             uint256 numContributions = contributions.length;
             for (uint256 i; i < numContributions; ++i) {
                 Contribution memory c = contributions[i];
-                if (c.previousTotalContributions >= totalEthUsed) {
+                if (c.previousTotalContributions - c.previousTotalContributionsWithdrawn >= totalEthUsed) {
```

Comment:
This doesn't really work. What I meant [here](https://github.com/PartyDAO/party-protocol/issues/193#issuecomment-1424819412) was that we'd subtract `totalContributionsWithdrawn` from `previousTotalContributions` when looping through each contribution. If doing it this way, you'd need to also include a condition for when `totalContributionsWithdrawn > previousTotalContributions`, which would mean the entire contribution was used.

Reply:
That's why I mentioned you'd add this condition:

> ... If doing it this way, you'd need to also include a condition for when `totalContributionsWithdrawn > previousTotalContributions`, which would mean the entire contribution was used.

Underflow isn't an issue here, if `totalContributionsWithdrawn > previousTotalContributions` we know the entire contribution was used.

Solution here should be to remove `previousTotalAmountWithdrawn` from the `Contribution` struct and instead subtract `totalContributionsWithdrawn` from `previousTotalContributions` for each contribution. There is no need to cast uints to ints. Just add handling for if `totalContributionsWithdrawn > previousTotalContributions` and we should be good.